plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -7895,8 +7895,8 @@ public void testOptimize()
         assertQueryFails(optimizeEnabledSession, "ALTER TABLE " + tableName + " EXECUTE \"optimize\"", "Procedure optimize not registered for catalog hive");
         assertUpdate(optimizeEnabledSession, "ALTER TABLE " + tableName + " EXECUTE \"OPTIMIZE\"");
         // optimize with delimited parameter name (and procedure name)
-        assertQueryFails(optimizeEnabledSession, "ALTER TABLE " + tableName + " EXECUTE \"OPTIMIZE\" (\"file_size_threshold\" => '10B')", "Catalog 'hive' table procedure 'OPTIMIZE' property '\"file_size_threshold\"' does not exist");
-        assertQueryFails(optimizeEnabledSession, "ALTER TABLE " + tableName + " EXECUTE \"OPTIMIZE\" (\"FILE_SIZE_THRESHOLD\" => '10B')", "Catalog 'hive' table procedure 'OPTIMIZE' property '\"file_size_threshold\"' does not exist");
+        assertUpdate(optimizeEnabledSession, "ALTER TABLE " + tableName + " EXECUTE \"OPTIMIZE\" (\"file_size_threshold\" => '10B')");
+        assertUpdate(optimizeEnabledSession, "ALTER TABLE " + tableName + " EXECUTE \"OPTIMIZE\" (\"FILE_SIZE_THRESHOLD\" => '10B')");
```

Comment:
This feels wrong. Only one of them should pass.

Reply:
It works like this:

- the distinction between delimited and not is first erased in https://github.com/trinodb/trino/blob/2051e78c7b38ce0e58a642bf2ac77beb9243ed55/core/trino-main/src/main/java/io/trino/metadata/TableProceduresPropertyManager.java#L90 which may be incidental
- then the case is normalized again at https://github.com/trinodb/trino/blob/9ee8f7415c8cd4f4e76a2e196ef2863f7872db9a/core/trino-main/src/main/java/io/trino/metadata/PropertyUtil.java#L60-L61

So
- if we want ALTER TABLE EXECUTE (...) properties to be case sensitive, both these places need to be changed/diverged
- do we want *all* properties to be case-sensitive? especially the table properties?

cc @martint @kasiafi @dain @losipiuk

Reply:
I would say that we only allow case-insensitive arguments for procedures. I don't see a benefit to allowing case-sensitive.

Reply:
i don't see a direct benefit here as well, and that's why @losipiuk and I decided to go accept case-insensitivity for ALTER TABLE EXECUTE's parameters.
However, there is a concern about coherence and conformance, which i'd leave up to @martint @kasiafi to debate.

Reply:
I think that we should respect the proper identifier semantics.
Although it might not seem very beneficial to allow using case-sensitive  arguments for procedures, it's about consistency with the spec, and consistency of behavior across Trino. When adding new features, we generally try to be consistent with the spec wrt identifiers. In the long term, we're going to apply per-spec identifiers everywhere.

Reply:
Filed https://github.com/trinodb/trino/issues/11326 to prevent scope creep in this PR.